Migrate IRGen tests to Swift 4

slavapestov opened the pull request.

slavapestov committed:
Migrate IRGen tests to Swift 4

slavapestov commented:
@swift-ci Please smoke test

jrose-apple commented on test/IRGen/dllimport.swift:
```diff
@@ -1,16 +1,16 @@
 
 public func get_ci() -> dllexport.c {
-  return dllexport.ci
+  return dllexport.ci.self
```
This one looks wrong (well, unnecessary).

slavapestov replied:
Yeah, ci is a _property_ not a type. Looks like we just ignore `.self` here, which seems like a bug.

jrose-apple replied:
No, you're allowed to write `.self` on anything. That's deliberate. But it's not necessary.

jrose-apple commented on test/IRGen/objc_extensions.swift:
```diff
@@ -34,24 +34,20 @@ import objc_extension_base
     return super.someMethod()
   }
-
-  // Overrides a class method of NSObject
```
Can you override a different class method instead? `isSubclass(of:)`, maybe?

slavapestov replied:
`isSubclass(of:)` does not appear to be overridable, but I defined another method in an extension and added an override, which has the same effect since `@objc` methods in extensions are dynamic.

jrose-apple replied:
:-/ That's not exactly the same test either, since the thing being overridden isn't imported. But I guess we have a lot of other tests for overrides.

slavapestov committed:
Re-add Swift 3 mangling test case

slavapestov commented:
@jrose-apple Thanks for the review feedback. I've updated the IRGen patch and added a new commit which brings back the mangling test you commented on in the SILGen patch.

slavapestov commented:
@swift-ci Please smoke test

slavapestov commented:
@swift-ci Please smoke test
